src/ORM/Marshaller.php

```diff
@@ -463,6 +466,9 @@ public function merge(EntityInterface $entity, array $data, array $options = [])
         $properties = $marshalledAssocs = [];
         foreach ($data as $key => $value) {
             if (!empty($errors[$key])) {
+                if (method_exists($entity, 'invalid')) {
```

Comment:
This will call `method_exists` for each property access. We should call the method once and store the results in a variable.